bashrc.d/zz-20-dev.sh

```diff
@@ -91,6 +91,13 @@ if [[ ! ${bivio_color:-} && $(type -p brew) ]]; then
     export HOMEBREW_NO_COLOR=1
 fi
 
+for r in 'sirepo'; do
+    p="$HOME/src/radiasoft/$r/etc/.home-env"
```

Comment:
quotes unnecessary and declare p. I don't think this should be a dot file, because people won't know it is there. It should also indicate the file type, e.g. home-env.sh.

There is a question about whether this should be done in all cases. In a different channel besides dev, this should not be read, because it can cause confusion with the environment. This presents a catch-22 which I'm not sure how we resolve, since the environment isn't set until after bashrc is read. One option is to create an environment variable that enables these functions. Another is to set rules that these are just functions, not environment, e.g. home-env-func.sh. 

Thoughts?

Reply:
One day I'll not quote the right side of assignments. Maybe time for me to use a linter...

Why declare p? I thought we did that for scoping and that only works inside functions?

Ya I was curious about the non dev cases too. I hadn't considered the problem of setting env which could be quite problematic. I don't have a good answer. 

Where would the variable be set? In post_bivio_bashrc? If so, it creates a discoverability problem and more headache when setting up to get the environment just right. I don't really like this option.

home-env-funcs.sh is fine. We could build into our CI to validate this?.

Reply:
9cdeb73